app/src/main/java/com/yuneec/example/component/fragment/GimbalFragment.java

```diff
@@ -95,32 +97,43 @@ public void onClick(View v) {
 
         switch (v.getId()) {
             case R.id.rotate_camera_clockwise:
-
-                if (editText.getText().toString().isEmpty()) {
-                    Toast.makeText(getActivity(), R.string.no_text, Toast.LENGTH_LONG).show();
+                Media.vibrate(getActivity());
+                if (yawVal.getText().toString().isEmpty()) {
+                    Common.makeToast(getActivity(), "Please enter yaw degree, before clicking rotate button");
                 } else {
                     try {
-                        float yaw_degree = Float.parseFloat(editText.getText().toString());
-                        Gimbal.asyncSetPitchAndYawOfJni(0, yaw_degree,
+                        float yaw_degree = Float.parseFloat(yawVal.getText().toString());
+                        Gimbal.asyncSetPitchAndYawOfJni(Common.currentPitch, yaw_degree,
                                                         GimbalListener.getGimbaListener());
-                        Common.currentRotation += Common.fixedRotationAngleDeg;
+                        Common.currentYaw = yaw_degree;
                     } catch (Exception e) {
-                        Toast.makeText(getActivity(), "Please enter a valid value for yaw degree",
-                                       Toast.LENGTH_LONG).show();
+                        Common.makeToast(getActivity(), "Please enter a valid value for yaw degree");
                     }
                 }
 
                 break;
-            /*case R.id.rotate_camera_anticlockwise:
-                Gimbal.asyncSetPitchAndYawOfJni(0, Common.currentRotation - Common.fixedRotationAngleDeg,
-                        GimbalListener.getGimbaListener());
-                Common.currentRotation -= Common.fixedRotationAngleDeg;
-                break;*/
             case R.id.rotate_to_initial:
+                Media.vibrate(getActivity());
                 Gimbal.asyncSetPitchAndYawOfJni(0, 0, GimbalListener.getGimbaListener());
-                Common.currentRotation = 0;
+                Common.currentYaw = 0;
+                Common.currentPitch = 0;
                 break;
+            case R.id.set_pitch:
```

Comment:
That's helpful.

On pitch, do you have a note that it's clear that the values are from [0..-90] ?

Reply:
Yes, I display it as a hint in the Edit Text field.